iptables/feature_detect.go

```diff
@@ -92,14 +103,15 @@ func (d *FeatureDetector) RefreshFeatures() {
 func (d *FeatureDetector) refreshFeaturesLockHeld() {
 	// Get the versions.  If we fail to detect a version for some reason, we use a safe default.
 	log.Debug("Refreshing detected iptables features")
-	iptV := d.getIptablesVersion()
+	iptV, iptMode := d.getIptablesVersion()
 	kerV := d.getKernelVersion()
 
 	// Calculate the features.
 	features := Features{
-		SNATFullyRandom:     iptV.Compare(v1Dot6Dot0) >= 0 && kerV.Compare(v3Dot14Dot0) >= 0,
-		MASQFullyRandom:     iptV.Compare(v1Dot6Dot2) >= 0 && kerV.Compare(v3Dot14Dot0) >= 0,
-		RestoreSupportsLock: iptV.Compare(v1Dot6Dot2) >= 0,
+		SNATFullyRandom:              iptV.Compare(v1Dot6Dot0) >= 0 && kerV.Compare(v3Dot14Dot0) >= 0,
+		MASQFullyRandom:              iptV.Compare(v1Dot6Dot2) >= 0 && kerV.Compare(v3Dot14Dot0) >= 0,
+		RestoreSupportsLock:          iptV.Compare(v1Dot6Dot2) >= 0,
+		IptablesRestoreHasReplaceBug: iptMode == "nft" && iptV.Compare(v1Dot8Dot2) <= 0,
```

Comment:
I know it's unrelated to your change, but... this code is quite hard to read, and I wonder if that's necessary.  I'd much rather read this:
```
		SNATFullyRandom:              iptV.AtLeast("1.6.0") && kerV.AtLeast("3.14.0"),
		MASQFullyRandom:              iptV.AtLeast("1.6.2") && kerV.AtLeast("3.14.0"),
		RestoreSupportsLock:          iptV.AtLeast("1.6.2"),
		IptablesRestoreHasReplaceBug: iptMode == "nft" && iptV.UpTo("1.8.2"),
```

Reply:
Agree, do you know of a lib that implements a nice API like that?  I just grabbed something off the shelf.

Reply:
I was thinking something like this:
```
func (v *Version) AtLeast(minVersion string) bool {
	return v.Compare(versionparse.MustParseVersion(minVersion) >= 0
}
```
I see that the Version type comes from github.com/hashicorp/go-version, but I think you can easily wrap that so as to permit adding another method, can't you?

Reply:
(Also, apologies for leaving this sitting for so long!)